airesources/OCaml/game.ml

```diff
@@ -0,0 +1,256 @@
+      ship.status = `Docked
+      && ship.docked_planet_id = planet.p_entity.id
+    ) state.all_ships
+  ) state.planets
+;;
+
+let update_map state =
+  state.turn <- state.turn + 1;
+  state.last_update <- Unix.gettimeofday ();
+  Debug.info (Printf.sprintf "--- Begin Turn %i ---\n" state.turn);
+  Networking.parse_map state (Networking.get_string());
+  set_planets_docked_ships state;
+;;
+
+let init state name initial_bot_setup =
+  state.my_id <- int_of_string (Networking.get_string ());
+  Debug.setup_logging state.my_id name;
+  Networking.read_width_height state (Networking.get_string ());
+  update_map state;
+  initial_bot_setup state;
+  Networking.send_string name;
+  Networking.done_sending (); 
+;;
+
+let send_command_queue =
+  Networking.send_moves 
+;;
+
+let get_me state =
+  state.player.(state.my_id)
+;;
+
+let get_player state id =
+  state.player.(id)
+;;
+
+let all_players state =
+  state.player
+;;
+
+let get_planet state id =
+  List.find (fun planet ->
+    planet.p_entity.id = id
+  ) state.planets
+;;
+
+let all_planets state =
+  state.planets
+;;
+
+let planet_entities planets = 
+  List.map (fun planet -> planet.p_entity) planets
+;;
+
+let ship_entities ships =
+  List.map (fun ship -> ship.s_entity) ships
+;;
+
+let all_ship_entities state =
+  ship_entities state.all_ships
+;;
+
+let all_entities state =
+  planet_entities state.planets
+  @ all_ship_entities state
+;;
+
+let all_entities_except state entity =
+  List.filter (fun e ->
+    not (e == entity)
+  ) (all_entities state)
+;;
+
+let calculate_distance_between e1 e2 =
+  sqrt ((e1.x -. e2.x) ** 2. +. (e1.y -. e2.y) ** 2.)
+;;
+
+let deg2rad v = (v -. 180. ) *. 6.28318530717958623 /. 360.;;
+
+let rad2deg v = ((v /. 6.28318530717958623) *. 360.) +. 180.;;
+
+let calculate_radians_between e1 e2 =
+  (atan2 (e2.y -. e1.y) (e2.x -. e1.x))
+;;
+
+let calculate_angle_between e1 e2 =
+  mod_float (rad2deg (atan2 (e2.y -. e1.y) (e2.x -. e1.x))) 360.
+;;
+
+let closest_point from_entity to_entity min_distance =
+  let angle = calculate_radians_between to_entity from_entity in
+  let radius = to_entity.radius +. min_distance in
+  let x = to_entity.x +. radius *. (cos angle) in
+  let y = to_entity.y +. radius *. (sin angle) in
+    x, y
+;;
+
+let nearby_entities_by_distance state entity =
+  List.map (fun e2 ->
+    e2, (calculate_distance_between entity e2)
+  ) (all_entities_except state entity)
+;;
+
+let all_ships state =
+  state.all_ships
+;;
+
+let docked_ships planet =
+  planet.docked_ships
+;;
+
+let get_planet_docked_ship planet ship_id =
+  try
+    Some (List.find (fun ship ->
+      ship.s_entity.id = ship_id
+    ) planet.docked_ships)
+  with _ -> None
+;;
+
+let planet_is_owned planet =
+  planet.p_entity.owner >= 0
+;;
+
+let planet_is_full planet =
+  List.length planet.docked_ship_ids >= planet.num_docking_spots
+;;
+
+let rec positive_int_degrees a =
+  if a < 0 then positive_int_degrees (a + 360)
```

Comment:
You can do this without a conditional: other kits with no modulo operator do it via ((x % 360) + 360) % 360.